### PR TITLE
E2E Different Resource Group

### PR DESCRIPTION
**Pull request by rpanackal**
## Context

AI/ai-sdk-java-backlog#123.

Include test to cover using OpenAI or Orchestration service with a different `resourceGroup` than `"default"`.

### Feature scope:
 
- [x] Add new endpoints in sample app for filtering completions by resource group
- [x] Fix bug in using custom destination from OpenAI client
  - [x] All calls to OpenAI service declares a default api version by default (unless explicitly set)

## Definition of Done

- [x] Functionality scope stated & covered
- [x] Tests cover the scope above
- [ ] ~Error handling created / updated & covered by the tests above~
- [ ] ~Aligned changes with the JavaScript SDK~
- [ ] ~Documentation updated~
- [ ] ~Release notes updated~

### CODE REVIEW

File: foundation-models/openai/src/test/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClientTest.java [hunk 1]
```diff
@@ -74,7 +74,9 @@ void apiVersion() {
     stubFor(post(anyUrl()).willReturn(okJson("{}")));
     Try.of(() -> client.chatCompletion(new OpenAiChatCompletionParameters()));
 
-    verify(exactly(1), postRequestedFor(anyUrl()).withoutQueryParam("api-version"));
+    verify(
+        exactly(1),
+        postRequestedFor(anyUrl()).withQueryParam("api-version", equalTo("2024-02-01")));
```
Comment (newtork) [inline foundation-models/openai/src/test/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClientTest.java:79]: (Explanation)

Without the `api-version` we experienced _resource not found_ error messages from AI Core.

File: sample-code/spring-app/pom.xml [hunk 2]
```diff
@@ -157,6 +157,10 @@
       <artifactId>assertj-core</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.sap.cloud.sdk.cloudplatform</groupId>
+      <artifactId>cloudplatform-connectivity</artifactId>
+    </dependency>
```
Comment (newtork) [inline sample-code/spring-app/pom.xml:163]: (Major)
Wait, why is this necessary?

**Review by newtork — APPROVED** [review]

File: sample-code/spring-app/pom.xml (hunk 2 above)
Comment (rpanackal) [reply sample-code/spring-app/pom.xml:163]: Without the declaration, the maven dependency plugin with goal "analyze-only" reports "Used undeclared dependencies found"

File: foundation-models/openai/src/test/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClientTest.java (hunk 1 above)
Comment (MatKuhr) [reply foundation-models/openai/src/test/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClientTest.java:79]: Correct, this was intended. The idea being that it is possible for users to prevent the api version from being set, e.g. for cases where the some model works without api version, or maybe it might be set automatically by long access; or if the key of that header ever changes.

That is why the javadoc of the method explicitly states to call withApiVersion afterwards 


But to be fair, the above are rather hypothetical scenarios and i think it may be worthwhile to change it in favor of the more convenient approach of setting this automatically

**Review by MatKuhr — APPROVED** [review]
Adding a small section in the docs for this would be useful, otherwise Lgtm
